packages/core/src/shared/lsp/manifestResolver.ts

```diff
@@ -90,25 +99,48 @@ export class ManifestResolver {
             return JSON.parse(content) as Manifest
         } catch (error) {
             throw new ToolkitError(
-                `Failed to parse manifest: ${error instanceof Error ? error.message : 'Unknown error'}`
+                `Failed to parse ${this.lsName} manifest: ${error instanceof Error ? error.message : 'Unknown error'}`
             )
         }
     }
 
+    /**
+     * Check if the current manifest is deprecated.
+     * If yes and user hasn't muted this notification, shows a toast message with two buttons:
+     * - OK: close and do nothing
+     * - Don't Show Again: Update global state (muteDeprecation) so the deprecation message is never shown for this manifest.
+     * @param manifest
+     */
     private checkDeprecation(manifest: Manifest): void {
+        const deprecationMessage = `${this.lsName} manifest is deprecated. No future updates will be available.`
         if (manifest.isManifestDeprecated) {
```

Comment:
Could probably invert this like:
```typescript
if (!manifest.isManifestDeprecated) {
    return
}

logger.info(deprecationMessage)
if (!this.getStorage()[this.lsName].muteDeprecation) {
    void vscode.window
        .showInformationMessage(deprecationMessage, localizedText.ok, localizedText.dontShow)
        .then((button) => {
            if (button === localizedText.dontShow) {
                this.getStorage()[this.lsName].muteDeprecation = true
            }
        })
}
```
instead of having a inner `ifs` nested